src/Nethermind/Nethermind.TxPool/ITxPool.cs

```diff
@@ -16,7 +15,7 @@ public interface ITxPool
         int GetPendingTransactionsCount();
         int GetPendingBlobTransactionsCount();
         Transaction[] GetPendingTransactions();
-        ConcurrentDictionary<byte[], List<Hash256>> GetBlobIndex();
+        IEnumerable<BlobAndProofV1?> GetBlobsAndProofs(byte[][] blobVersionedHashes);
```

Comment:
I think we are messing abstraction layers here:
1. TxPool doesn't need to handle multiple searches directly - would only do one, and handle collections outside of it. 
2. `BlobAndProofV1` is typical RPC type and should not be used in TxPool directly

So it can look like:  `bool TryGetBlob([NotNullWhen(true)] out byte[]? blob, [NotNullWhen(true)] out byte[]? proof)`